### PR TITLE
Add new WDK record scope: record-collapsed

### PR DESCRIPTION
**Pull request by bobular**
See https://github.com/VEuPathDB/OrthoMCLModel/commit/23305fb7457254d80c05198a6c346d9a24ce3e41 where an example configuration change has been made on the `diamond` branch.

Here we handle it on the client side.

To do:
- [ ] documentation

### CODE REVIEW

File: packages/libs/wdk-client/src/StoreModules/RecordStoreModule.ts [hunk 1]
```diff
@@ -84,11 +84,22 @@ export function reduce(state: State = {} as State, action: Action): State {
...
+
+      const collapsedSections = getLeaves(categoryTree, (node) => node.children)
+        .filter(
+          (
+            node
+          ): node is CategoryTreeNode & { properties: { name: string[] } } =>
+            !!node.properties.scope?.includes('record-collapsed') &&
+            !!node.properties.name
+        )
+        .map((node) => node.properties.name[0]);
```
Comment (bobular) [inline packages/libs/wdk-client/src/StoreModules/RecordStoreModule.ts:96]: Is there ever >1 element in the `name` array?

File: packages/libs/wdk-client/src/Controllers/RecordController.tsx [hunk 2]
```diff
@@ -111,7 +111,7 @@ class RecordController extends PageController<Props> {
           return (
             attributes?.includes(id) ||
             tables?.includes(id) ||
-            scopes.includes('record-internal')
+            scopes.includes('record-internal') // do we need to handle 'record-collapsed'?
```
Comment (dmfalke) [inline packages/libs/wdk-client/src/Controllers/RecordController.tsx:114]: No, it isn't needed here. This code comes into play when attributes or tables are specified in the url, which is a way to have a record page with only those attributes and tables. Clearly this needs to be better documented.

File: packages/libs/wdk-client/src/StoreModules/RecordStoreModule.ts (hunk 1 above)
Comment (dmfalke) [reply packages/libs/wdk-client/src/StoreModules/RecordStoreModule.ts:96]: I don't think so, but the ontology properties are always multi-value.

FYI, you can use this function:
https://github.com/VEuPathDB/web-monorepo/blob/c02d237d2116a452a8195a51ad570b10e6ccdb2e/packages/libs/wdk-client/src/Utils/CategoryUtils.tsx#L91-L93

**Review by dmfalke — APPROVED** [review]
This looks great!

**Pull request merged**